libdnf/repo/repo_sack.cpp

```diff
@@ -84,6 +88,93 @@ RepoWeakPtr RepoSack::get_cmdline_repo() {
 }
 
 
+std::map<std::string, libdnf::rpm::Package> RepoSack::add_cmdline_packages(const std::vector<std::string> & paths) {
+    // find remote URLs and local file paths in the input
+    std::vector<std::string> rpm_urls;
+    std::vector<std::string> rpm_filepaths;
+    const std::string_view ext(".rpm");
+    for (const auto & spec : paths) {
+        if (libdnf::utils::url::is_url(spec)) {
+            rpm_urls.emplace_back(spec);
+        } else if (spec.length() > ext.length() && spec.ends_with(ext)) {
+            rpm_filepaths.emplace_back(spec);
+        }
+    }
+
+    if (rpm_urls.empty() && rpm_filepaths.empty()) {
+        // nothing to fill into cmdline repo
+        return {};
+    }
+
+    auto cmdline_repo = get_cmdline_repo();
+
+    // Ensure that download location for command line repository exists
+    std::filesystem::path cmd_repo_pkgs_dir{cmdline_repo->get_cachedir()};
+    cmd_repo_pkgs_dir /= CACHE_PACKAGES_DIR;
+    std::filesystem::create_directories(cmd_repo_pkgs_dir);
+
+    // map remote URLs to local destination files
+    std::map<std::string, std::filesystem::path> url_to_path;
```

Comment:
Minor: What about use `std::string_view` to save some memory?
`std::map<std::string_view, std::filesystem::path> url_to_path;`
and use `std::string{url}` on line 153.

Reply:
Done.